include/engine/base64.hpp

```diff
@@ -32,18 +47,24 @@ namespace engine
 // Encodes a chunk of memory to Base64.
 inline std::string encodeBase64(const unsigned char *first, std::size_t size)
 {
-    using namespace boost::archive::iterators;
+    std::vector<unsigned char> bytes{first, first + size};
+    BOOST_ASSERT(!bytes.empty());
 
-    const std::string bytes{first, first + size};
+    std::size_t bytes_to_pad{0};
 
-    using Iter = base64_from_binary<transform_width<std::string::const_iterator, 6, 8>>;
+    while (bytes.size() % 3 != 0)
+    {
```

Comment:
I would prefer some note on why we use 3 here, since the connection to the 6/8 part is not obvious.